examples/jsm/utils/BufferGeometryUtils.js

```diff
@@ -720,7 +722,7 @@ function mergeVertices( geometry, tolerance = 1e-4 ) {
 
 		const tmpAttribute = tmpAttributes[ name ];
 
-		result.setAttribute( name, new BufferAttribute(
+		result.setAttribute( name, new tmpAttribute.constructor(
```

Comment:
Replaced everywhere in mergeVertices `new BufferAttribute` with `new <oldThing>.constructor` to preserve eventual optimization that the specialized class might have.

Reply:
If someone would use `Float16BufferAttribute`, the new approach is required in any event.